File: DSA/Abhinn_Tyagi_500121872/Day32_Question_3.c
#include <stdio.h>

int xorAll(int arr[], int n) {
    int result = 0;
    for (int i = 0; i < n; i++) {
        result ^= arr[i];  
    }
    return result;
}

int main() {
    int arr[] = {1, 2, 3};
    int n = sizeof(arr) / sizeof(arr[0]);

    int result = xorAll(arr, n);
    printf("XOR of all elements: %d\n", result);
    return 0;
}

File: DSA/VaniJain_590012020/Day32Question3.java
import java.util.*;
public class ArrayXOR 
{
    public static int findXOR(int[] arr) 
  {
        int xor = 0;
        for (int num : arr) 
        {
            xor ^= num;
        }
        return xor;
    }
    public static void main(String[] args) 
  {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter size of array: ");
        int n = sc.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter " + n + " elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        int result = findXOR(arr);
        System.out.println("XOR of all elements in the array is: " + result);
    }
}
